### DISCUSSION TITLE
Using virtual environments on a self-hosted runner?

### DISCUSSION THREAD

**Discussion by vadi2** [actions/virtual-environments#3511] [Installed software]
I'm using a self-hosted runner only because Github isn't yet able to provide an M1 chip, and the disadvantage of that is that I have to configure all the software myself. I have no intention of maintaining the software myself.

Is it possible to use the virtual environment on my own M1 mac?

**Comment by maxim-lobanov** [comment]
Hello @vadi2 , unfortunately, it is not possible for now.
We don't build images for M1 for now because we don't have infrastructure for that. Also we can't distribute ready images due to legal issues.

**Reply by vadi2** [reply]
Thanks for the reply!

**Reply by tuler** [reply]
Hi @maxim-lobanov can you elaborate on the legal issues?

**Reply by panekj** [reply]
@tuler they install software which licence prohibits redistribution

**Reply by tuler** [reply]
Which software is he referring to? The GitHub runner has been [released with Apple M1 support](https://github.com/actions/runner/releases/tag/v2.292.0).

**Reply by panekj** [reply]
> Which software is he referring to?

Software that is installed as part of this repository.

> The GitHub runner has been [released with Apple M1 support](https://github.com/actions/runner/releases/tag/v2.292.0).

This is just GHA agent.

**Reply by tuler** [reply]
I see. So when he says "Also we can't distribute ready images due to legal issues" it's not M1 specific, it's for every platform.

**Reply by panekj** [reply]
Yes, as mentioned in https://github.com/actions/virtual-environments/issues/176#issuecomment-613117622 and https://github.com/actions/virtual-environments/issues/3428#issuecomment-845143194

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
